src/hotspot/share/gc/g1/g1HeapRegionManager.cpp

```diff
@@ -478,7 +478,7 @@ uint G1HeapRegionManager::find_contiguous_in_free_list(uint num_regions) {
 
 uint G1HeapRegionManager::find_contiguous_allow_expand(uint num_regions) {
   // Check if we can actually satisfy the allocation.
-  if (num_regions > available()) {
+  if (num_regions > (num_free_regions() + available())) {
```

Comment:
The name "available" is too vague -- without looking at its impl, one could think free regions should be "available" as well. I wonder if sth like "num_uncommitted" is more precise.

Reply:
This seems to be a pre-existing issue. Filed https://bugs.openjdk.org/browse/JDK-8355976.

Afair this member has always been called that way.

Reply:
Yeah, I tripped up on that too.